### ISSUE TITLE
Add option to trim to nearest paragraph or sentence

### ISSUE THREAD

**Issue by ghost** [backdrop-contrib/smart_trim#10]
It'd be nice to have the option to trim to the nearest paragraph or sentence.

For example, instead of:

```html
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean consectetur purus in sem suscipit pharetra. Mauris a vestibulum massa. Suspendisse tempus felis ut tincidunt porta. Nunc lorem nulla, aliquam ut pretium imperdiet, interdum ac ipsum.</p>
<h2>A heading</h2>
<p>Integer et molestie est. Curabitur...</p>
```

You could have:

```html
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean consectetur purus in sem suscipit pharetra. Mauris a vestibulum massa. Suspendisse tempus felis ut tincidunt porta. Nunc lorem nulla, aliquam ut pretium imperdiet, interdum ac ipsum....</p>
```

Here're Drupal issues requesting the sentence option (there seem to be a few un-marked duplicates with their own patches): https://www.drupal.org/project/issues/smart_trim?text=sentence&status=All&priorities=All&categories=All&version=7.x&component=All
And here's one for paragraphs: https://www.drupal.org/project/smart_trim/issues/2063955

**Comment by ghost** [comment]
Just realised that Smart Trim doesn't even allow you to trim to the nearest word... What it does allow is specifying whether the Trim Length is counted as characters or words.

Actually, if Views is available, it uses the `views_trim_text()` function and hard-codes the `word_boundary` option to `TRUE`. If not, it uses the `text_summary()` function which automatically tries to break "at a sensible place such as the end of a paragraph, a line break, or the end of a sentence (in that order of preference)." However, since Views is a required module, it's always available, so `text_summary()` is never actually used.

So, in other words, what I'd like to be able to do is specify a trim length of, say, 500 characters, but to have it trim to the nearest paragraph below that (possibly specifying a minimum length too, in case the nearest paragraph break below 500 characters is just one sentence, in which case it'd break at nearest sentence below 500 chars instead).
